src/main/java/uk/ac/ebi/eva/pipeline/io/readers/VcfReader.java

```diff
@@ -0,0 +1,253 @@
+        for (VCFHeaderLine vcfHeaderLine : headerLineListEntry.getValue()) {
+            if (vcfHeaderLine instanceof VCFSimpleHeaderLine) {
+                VCFSimpleHeaderLine line = ((VCFSimpleHeaderLine) vcfHeaderLine);
+                VcfAlternateHeader biodataLine = new VcfAlternateHeader(line.toString());
+                vcfHeaders.add(biodataLine);
+            }
+        }
+        return vcfHeaders;
+    }
+
+    private List<VcfFormatHeader> getVcfFormatHeaders(Map.Entry<String, List<VCFHeaderLine>> headerLineListEntry) {
+        List<VcfFormatHeader> vcfHeaders = new ArrayList<>();
+        for (VCFHeaderLine vcfHeaderLine : headerLineListEntry.getValue()) {
+            if (vcfHeaderLine instanceof VCFFormatHeaderLine) {
+                VCFFormatHeaderLine line = ((VCFFormatHeaderLine) vcfHeaderLine);
+                VcfFormatHeader biodataLine = new VcfFormatHeader(line.getID(), getNumber(line),
+                                                                  line.getType().toString(),
+                                                                  line.getDescription());
+                vcfHeaders.add(biodataLine);
+            }
+        }
+        return vcfHeaders;
+    }
+
+    private List<VcfFilterHeader> getVcfFilterHeaders(Map.Entry<String, List<VCFHeaderLine>> headerLineListEntry) {
+        List<VcfFilterHeader> vcfHeaders = new ArrayList<>();
+        for (VCFHeaderLine vcfHeaderLine : headerLineListEntry.getValue()) {
+            if (vcfHeaderLine instanceof VCFFilterHeaderLine) {
+                VCFFilterHeaderLine line = ((VCFFilterHeaderLine) vcfHeaderLine);
+                VcfFilterHeader biodataLine = new VcfFilterHeader(line.toString());
+                vcfHeaders.add(biodataLine);
+            }
+        }
+        return vcfHeaders;
+    }
+
+    private List<VcfInfoHeader> getVcfInfoHeaders(Map.Entry<String, List<VCFHeaderLine>> headerLineListEntry) {
+        List<VcfInfoHeader> vcfHeaders = new ArrayList<>();
+        for (VCFHeaderLine vcfHeaderLine : headerLineListEntry.getValue()) {
+            if (vcfHeaderLine instanceof VCFInfoHeaderLine) {
+                VCFInfoHeaderLine line = ((VCFInfoHeaderLine) vcfHeaderLine);
+                VcfInfoHeader biodataLine = new VcfInfoHeader(line.getID(), getNumber(line), line.getType().toString(),
+                                                              line.getDescription());
+                vcfHeaders.add(biodataLine);
+            }
+        }
+        return vcfHeaders;
+    }
+
+    public String getNumber(VCFCompoundHeaderLine line) {
+        String number;
+        switch (line.getCountType()) {
+            case A:
+                number = VCFConstants.PER_ALTERNATE_COUNT;
+                break;
+            case R:
+                number = VCFConstants.PER_ALLELE_COUNT;
+                break;
+            case G:
+                number = VCFConstants.PER_GENOTYPE_COUNT;
+                break;
+            case UNBOUNDED:
```

Comment:
yes, https://samtools.github.io/htsjdk/javadoc/htsjdk/constant-values.html#htsjdk.variant.vcf.VCFConstants.UNBOUNDED_ENCODING_v4. The API returns -1 for unknown number, but that constant is what's used for that purpose: [docs](https://samtools.github.io/htsjdk/javadoc/htsjdk/htsjdk/variant/vcf/VCFCompoundHeaderLine.html#getCount-htsjdk.variant.variantcontext.VariantContext-), [implementation](https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/variant/vcf/VCFCompoundHeaderLine.java#L85)